tests/integration/targets/aci_aaa_certificate_authority/tasks/main.yml

```diff
@@ -0,0 +1,179 @@
+# Test code for the ACI modules
+# Copyright: (c) 2024, Akini Ross (@akinross) <akinross@cisco.com>
+#
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  ansible.builtin.fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  ansible.builtin.set_fact:
+    aci_info: &aci_info
+      host: "{{ aci_hostname }}"
+      username: "{{ aci_username }}"
+      password: "{{ aci_password }}"
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: '{{ aci_use_proxy | default("debug") }}'
+
+# CLEAN ENVIRONMENT
+- name: Clean certificate_authority_1
+  cisco.aci.aci_aaa_certificate_authority: &certificate_authority_1_absent
+    <<: *aci_info
+    name: certificate_authority_1
+    state: absent
+
+- name: Clean certificate_authority_2
+  cisco.aci.aci_aaa_certificate_authority: &certificate_authority_2_absent
+    <<: *aci_info
+    name: certificate_authority_2
+    state: absent
+  register: result
+  until: result is not failed
```

Comment:
What is the purpose of using until and retries here?

How does the combination of until and retries work here? If result is still fail at retry number 4, does the task fail?

Reply:
certificate authority configure restarts a server in the background handling request, so repeat is needed when making changes to the certificate authority

correct after 4 tries the task fails indeed